### PR TITLE
Feature: Disable switches for background audio and tile animations

### PR DESCRIPTION
**Pull request by pizkaz**
This is the single remaining functional feature of the XCE fork (except for interactive layers, which can be implemented through the scripting API on a per-map basis).

It adds two checkboxes in "Other settings":
* "Block ambient sounds and music" ignores `PLAY_AUDIO` map properties.
* "Disable map tile animations" does exactly what it says.

Both settings are persisted in local storage. Translations are incomplete.

Supersedes https://github.com/workadventure/workadventure/pull/1691

### CODE REVIEW

File: play/src/i18n/ca-ES/menu.ts [hunk 1]
```diff
@@ -56,6 +56,8 @@ const menu: DeepPartial<Translation["menu"]> = {
         notifications: "Notificacions",
         cowebsiteTrigger: "Preguntar sempre abans d'obrir llocs web i habitacions Jitsi Meet",
         ignoreFollowRequest: "Ignorar peticions de seguir altres usuaris",
+        blockAudio: "Block ambient sounds and music",
+        disableAnimations: "Disable map tile animations",
```
Comment (moufmouf) [inline play/src/i18n/ca-ES/menu.ts:60]:
```suggestion
```

Best thing is to leave translations empty if not available. The translation library will fallback to English anyway!

File: play/src/i18n/es-ES/menu.ts [hunk 2]
```diff
@@ -56,6 +56,8 @@ const menu: DeepPartial<Translation["menu"]> = {
         notifications: "Notificaciones",
         cowebsiteTrigger: "Preguntar siempre antes de abrir sitios web y habitaciones Jitsi Meet",
         ignoreFollowRequest: "Ignorar peticiones de seguir a otros usuarios",
+        blockAudio: "Block ambient sounds and music",
+        disableAnimations: "Disable map tile animations",
```
Comment (moufmouf) [inline play/src/i18n/es-ES/menu.ts:60]:
```suggestion
```

File: play/src/i18n/pt-BR/menu.ts [hunk 3]
```diff
@@ -56,6 +56,8 @@ const menu: DeepPartial<Translation["menu"]> = {
         notifications: "Notificações",
         cowebsiteTrigger: "Sempre pergunte antes de abrir sites e salas do Jitsi Meet",
         ignoreFollowRequest: "Ignorar solicitações para seguir outros usuários",
+        blockAudio: "Block ambient sounds and music",
+        disableAnimations: "Disable map tile animations",
```
Comment (moufmouf) [inline play/src/i18n/pt-BR/menu.ts:60]:
```suggestion
```

File: play/src/i18n/zh-CN/menu.ts [hunk 4]
```diff
@@ -56,6 +56,8 @@ const menu: DeepPartial<Translation["menu"]> = {
         notifications: "通知",
         cowebsiteTrigger: "在打开网页和Jitsi Meet会议前总是询问",
         ignoreFollowRequest: "忽略跟随其他用户的请求",
+        blockAudio: "Block ambient sounds and music",
+        disableAnimations: "Disable map tile animations",
```
Comment (moufmouf) [inline play/src/i18n/zh-CN/menu.ts:60]:
```suggestion
```

**Review by moufmouf — COMMENTED** [review]
Thanks a lot for this PR!

Do you think you could add a small E2E test for the audio blocker testing that the audio volume bar is indeed hidden when the "block audio" checkbox is checked and someone enters an audio zone?

**Comment by pizkaz** [comment]
> Do you think you could add a small E2E test for the audio blocker testing that the audio volume bar is indeed hidden when the "block audio" checkbox is checked and someone enters an audio zone?

So far, I have not looked into E2E tests. I can try. :smiley_cat:

File: play/src/i18n/ca-ES/menu.ts (hunk 1 above)
Comment (pizkaz) [reply play/src/i18n/ca-ES/menu.ts]: Done.

File: play/src/i18n/pt-BR/menu.ts (hunk 3 above)
Comment (pizkaz) [reply play/src/i18n/pt-BR/menu.ts]: Done.

File: play/src/i18n/zh-CN/menu.ts (hunk 4 above)
Comment (pizkaz) [reply play/src/i18n/zh-CN/menu.ts]: Done.

File: play/src/i18n/es-ES/menu.ts (hunk 2 above)
Comment (pizkaz) [reply play/src/i18n/es-ES/menu.ts]: Done.

**Comment by pizkaz** [comment]
Hmm, something like this maybe?

The test fails with:
`Error: Unable to find the script frame. Is there one defined on the map?`

I suspect the map should have a `script` property. Should I create a new map for this test or edit the one I'm using here?

**Comment by pizkaz** [comment]
I think I got it. \o/

**Comment by moufmouf** [comment]
You nailed it! Awesome! Thanks so much :heart:

**Pull request merged**